darts/utils/historical_forecasts/optimized_historical_forecasts.py

```diff
@@ -137,22 +143,26 @@ def _optimized_historical_forecasts_regression_last_points_only(
                 :,
             ]
 
-        forecasts_list.append(
-            TimeSeries.from_times_and_values(
-                times=times[0]
-                if stride == 1 and model.output_chunk_length == 1
-                else generate_index(
-                    start=hist_fct_start + (forecast_horizon - 1) * freq,
-                    length=forecast.shape[0],
-                    freq=freq * stride,
-                    name=series_.time_index.name,
-                ),
-                values=forecast,
-                columns=forecast_components,
-                static_covariates=series_.static_covariates,
-                hierarchy=series_.hierarchy,
-            )
+        forecast_value = TimeSeries.from_times_and_values(
+            times=times[0]
+            if stride == 1 and model.output_chunk_length == 1
```

Comment:
Is it not better to do this in one more step? Because in this way it is really difficult to read and understand this method call...

Reply:
You're right, now everything is done in one go without creating TimeSeries manually